mcpserver/tools.go

```diff
@@ -194,7 +220,7 @@ func EnvironmentInfoToCallResult(envInfo *environment.EnvironmentInfo) (*mcp.Cal
 	return mcp.NewToolResultText(out), nil
 }
 
-var EnvironmentOpenTool = &Tool{
+var EnvironmentOpenTool = &Tool[Request, Response]{
```

Comment:
🔧 **Maintainability (Low):** EnvironmentOpenTool uses generic Request/Response types instead of specific typed requests like the converted tools. Consider consistency in approach.

```suggestion
var EnvironmentOpenTool = &Tool[BaseRepositoryRequest, *environment.Environment]{
```